### PR TITLE
Repository & refactor ProductController

### PR DESCRIPTION
**Pull request by khieutrung12**
## Related Tickets:

-   [#47473](https://edu-redmine.sun-asterisk.vn/projects/ecommerce-intern-hn/issues/47473)

## WHAT this PR does?

Create repository & refactor ProductController

## Notes Impacted Areas:

-   N/A.

## Attachments:

-   N/A.

### CODE REVIEW

**Comment by khieutrung12** [comment]
ready

File: app/Repositories/Product/ProductRepository.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        return $this->model->select('*')->with('brand', 'category')->orderby('created_at', 'DESC')->get();
+    }
+
+    public function createProduct(array $attributes, array $categoryIds)
+    {
+        $product = $this->create($attributes);
+        $product->category()->sync($categoryIds);
+        $product->save();
+        return $product;
```
Comment (huyenttm-0901) [inline app/Repositories/Product/ProductRepository.php]: convention

File: app/Repositories/Product/ProductRepository.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function updateProduct(int $productId, array $attributes, array $categoryIds)
+    {
+        $product = $this->update($productId, $attributes);
+        $product->category()->sync($categoryIds);
+        $product->save();
+        return $product;
```
Comment (huyenttm-0901) [inline app/Repositories/Product/ProductRepository.php]: tương tự

**Comment by khieutrung12** [comment]
ready

**Review by DTHNhung — APPROVED** [review]

**Review by huyenttm-0901 — COMMENTED** [review]
chưa fix comment

File: app/Repositories/Product/ProductRepository.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return $this->model->select('*')->with('brand', 'category')->orderby('created_at', 'DESC')->get();
+    }
+
+    public function createProduct(array $attributes, array $categoryIds)
+    {
+        $product = $this->create($attributes);
+        $createProduct = $product;
+        $product->category()->sync($categoryIds);
+        $product->save();
+        return $createProduct;
```
Comment (huyenttm-0901) [inline app/Repositories/Product/ProductRepository.php]: vẫm chưa fix comment này em

File: app/Repositories/Product/ProductRepository.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public function updateProduct(int $productId, array $attributes, array $categoryIds)
+    {
+        $product = $this->update($productId, $attributes);
+        $updateProduct = $product;
+        $product->category()->sync($categoryIds);
+        $product->save();
+        return $updateProduct;
```
Comment (huyenttm-0901) [inline app/Repositories/Product/ProductRepository.php]: đây nữa

**Comment by khieutrung12** [comment]
ready

**Review by huyenttm-0901 — APPROVED** [review]

**Pull request merged**